test/mocks/mocks.go

```diff
@@ -373,8 +375,9 @@ func ServiceBindingRequestMock(
 	applicationResourceRef string,
 	matchLabels map[string]string,
 	bindUnannotated bool,
+	backendGvk *metav1.GroupVersionKind,
```

Comment:
Since all the information is coming from the caller anyway, why not leave the responsibility to fix the `BackingServiceSelector` field to the caller itself? For example, a SBR mock could be augmented through composition instead:

```go
// create a new mock
sbr := ServiceBindingRequestMock(...)
// configure the SBR's backing service selector
setBackingServiceSelectorMock(backingServiceResourceRef, backendGVK)
```

Dealing with the differences in the caller site is fine in this case, and then only on required places this function would be called to modify the SBR available, instead of introducing a new parameter and setting `nil` in almost every call site, which makes me like this parameter approach less.

Reply:
@isutton makes sense

Reply:
@isutton Introduced dedicated function to modify SBR